src/averageAges.js

```diff
@@ -36,8 +42,23 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
+
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let women = people.filter(person => person.sex === 'f');
+
+  if (withChildren) {
+    const mothersList = new Set(people.map(person => person.mother));
+
+    mothersList.delete(null);
```

Comment:
you're deleting `null` but reading a property from an object (`person.mother`) that doesn't exist return `undefined`